core/src/main/java/com/datastrato/gravitino/catalog/CatalogManager.java

```diff
@@ -585,6 +585,43 @@ private CatalogWrapper createCatalogWrapper(CatalogEntity entity) {
     return wrapper;
   }
 
+  private Set<String> getHiddenPropertyNames(CatalogEntity entity) {
+    Map<String, String> conf = entity.getProperties();
+    String provider = entity.getProvider();
+
+    IsolatedClassLoader classLoader = createClassLoader(provider, conf);
+    BaseCatalog<?> catalog = createBaseCatalog(classLoader, entity);
+
+    return classLoader.withClassLoader(
+        cl -> {
+          PropertiesMetadata catalogPropertiesMetadata = catalog.ops().catalogPropertiesMetadata();
+          return catalogPropertiesMetadata.propertyEntries().values().stream()
+              .filter(PropertyEntry::isHidden)
+              .map(PropertyEntry::getName)
+              .collect(Collectors.toSet());
+        },
+        RuntimeException.class);
+  }
```

Comment:
Shall we close the classloader here?

Reply:
+1. 
Why don't you get the class loader from the cache and use it? The current implementation will have a resource leak risk.

Reply:
> Why don't you get the class loader from the cache and use it? 

Since `listCatalogInfo` will list all catalogs, it will always refresh the cache

> The current implementation will have a resource leak risk.

try-with-resource added